### ISSUE TITLE
flair ner on custom data fails to start training and gets stuck before 1st epoch

### ISSUE THREAD

**Issue by anitchakraborty** [flairNLP/flair#2540]
I'm trying to train a custom NER with falir using my data as the BIO tagged format. 

Sample data in df to dict format:
```
[{'text': {0: 'MAINETTI', 1: '(', 2: 'Bangladesh', 3: ')', 4: 'PVT'},
 'labels': {0: 'B-ORG', 1: 'I-ORG', 2: 'I-ORG', 3: 'I-ORG', 4: 'I-ORG'}}]
```

I have created training and dev and test set accordingly in BIO format.

Training script:
```
from flair.data import Corpus
from flair.datasets import ColumnCorpus

# define columns
columns = {0: 'text', 1: 'ner'}
data_folder = 'TDC/'
corpus: Corpus = ColumnCorpus(data_folder, columns,
                              train_file='train.txt',
                              test_file='test.txt',
                              dev_file='dev.txt')

# tag to predict
tag_type = 'ner'
# make tag dictionary from the corpus
tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)

from flair.embeddings import WordEmbeddings, StackedEmbeddings, TokenEmbeddings
from typing import List

embedding_types : List[TokenEmbeddings] = [
        WordEmbeddings('glove'),
        ## other embeddings
        ]
embeddings : StackedEmbeddings = StackedEmbeddings(
                                 embeddings=embedding_types)
from flair.models import SequenceTagger
tagger : SequenceTagger = SequenceTagger(hidden_size=256,
                                       embeddings=embeddings,
                                       tag_dictionary=tag_dictionary,
                                       tag_type=tag_type,
                                       use_crf=True)
from flair.trainers import ModelTrainer
trainer : ModelTrainer = ModelTrainer(tagger, corpus)
trainer.train('resources/taggers/example-ner',
              learning_rate=0.1,
              mini_batch_size=32,
              max_epochs=150)
```
Now everything works find and started the training but it doesn't even reach the epoch one and stuck before the epochs in this stage:
```
2021-12-03 18:45:45,730 ----------------------------------------------------------------------------------------------------
2021-12-03 18:45:45,732 Model: "SequenceTagger(
  (embeddings): StackedEmbeddings(
    (list_embedding_0): WordEmbeddings(
      'glove/glove.txt'
      (embedding): Embedding(400001, 50)
    )
  )
  (word_dropout): WordDropout(p=0.05)
  (locked_dropout): LockedDropout(p=0.5)
  (embedding2nn): Linear(in_features=50, out_features=50, bias=True)
  (rnn): LSTM(50, 256, batch_first=True, bidirectional=True)
  (linear): Linear(in_features=512, out_features=8, bias=True)
  (beta): 1.0
  (weights): None
  (weight_tensor) None
)"
2021-12-03 18:45:45,733 ----------------------------------------------------------------------------------------------------
2021-12-03 18:45:45,734 Corpus: "Corpus: 1 train + 1 dev + 1 test sentences"
2021-12-03 18:45:45,736 ----------------------------------------------------------------------------------------------------
2021-12-03 18:45:45,737 Parameters:
2021-12-03 18:45:45,738  - learning_rate: "0.1"
2021-12-03 18:45:45,739  - mini_batch_size: "32"
2021-12-03 18:45:45,740  - patience: "3"
2021-12-03 18:45:45,742  - anneal_factor: "0.5"
2021-12-03 18:45:45,743  - max_epochs: "150"
2021-12-03 18:45:45,744  - shuffle: "True"
2021-12-03 18:45:45,746  - train_with_dev: "False"
2021-12-03 18:45:45,747  - batch_growth_annealing: "False"
2021-12-03 18:45:45,749 ----------------------------------------------------------------------------------------------------
2021-12-03 18:45:45,750 Model training base path: "resources\taggers\example-ner"
2021-12-03 18:45:45,751 ----------------------------------------------------------------------------------------------------
2021-12-03 18:45:45,753 Device: cpu
2021-12-03 18:45:45,754 ----------------------------------------------------------------------------------------------------
2021-12-03 18:45:45,756 Embeddings storage mode: cpu
2021-12-03 18:45:45,768 ----------------------------------------------------------------------------------------------------
```
There's no memory throttle as the RAM is 250 GB. After 4 hours, no epoch started and all the loss files and weight.txt files are created under the resources\taggers\example-ner directory but all of them are empty. Is there anything wrong with my process as no errors reflecting and the training hasn't started.

**Comment by whoisjones** [comment]
hi @anitchakraborty, from your script provided I assume that you need to change the tag_type to 'labels'. Our column corpus class checks for the respective column name which is usually 'ner', i.e. in `CONLL_03`. In your data snippet it looks like the column name is `labels`.

if this does not work, feel free to share a short data extract for us to reproduce.

**Comment by iliaNecrov** [comment]
Having the same issue, have found out the solutuion, @anitchakraborty

**Comment by elazzouzi1080** [comment]
Having the same issue?
